runner/operator/tempo_nextflow_operator/tempo_nextflow_operator.py

```diff
@@ -0,0 +1,410 @@
+            num_ns = len(normal_samples)
+            num_ts = len(tumor_samples)
+            if num_ns != num_ts:
+                print("Number of tumors and normals not the same; can't pair")
+            else:
+                for i in range(0, num_ns):
+                    tumor_id = tumor_samples[i]
+                    normal_id = normal_samples[i]
+                    pre_pairing[tumor_id] = normal_id
+        patient_ids = set()
+        patient_files = dict()
+        no_patient_samples = list()
+        for entry in tempo_files:
+            patient_id = entry.metadata[settings.PATIENT_ID_METADATA_KEY]
+            if patient_id:
+                patient_ids.add(patient_id)
+                if patient_id not in patient_files:
+                    patient_files[patient_id] = list()
+                patient_files[patient_id].append(entry)
+            else:
+                no_patient_samples.append(entry)
+
+        self.patients = dict()
+        self.non_cmo_patients = dict()
+        for patient_id in patient_files:
+            if "C-" in patient_id[:2]:
+                self.patients[patient_id] = patient_obj.Patient(patient_id, patient_files[patient_id], pre_pairing)
+            else:
+                self.non_cmo_patients[patient_id] = patient_obj.Patient(patient_id, patient_files[patient_id])
+
+        # output these strings to file
+        # input_json["conflict_data"] = self.create_conflict_samples_txt_file()
+        # input_json["unpaired_data"] = self.create_unpaired_txt_file()
+        mapping_all = self.create_mapping_input()
+        pairing_all = self.create_pairing_input()
+
+        beagle_version = __version__
+        run_date = datetime.now().strftime("%Y%m%d_%H:%M:%f")
+        tags = {"beagle_version": beagle_version, "run_date": run_date}
+
+        tumors = FileRepository.filter(
+            metadata={settings.REQUEST_ID_METADATA_KEY: self.request_id, "tumorOrNormal": "Tumor"},
+            values_metadata="ciTag",
+        )
+
+        jobs = []
+        for tumor in tumors:
+            pairing = self.get_pairing_for_sample(tumor, pairing_all)
+            mapping = self.get_mapping_for_sample(tumor, pairing["normal"], mapping_all)
+
+            name = "Tempo Run T:{tumor}, N:{normal}: {run_date}".format(
+                tumor=pairing["tumor"], normal=pairing["normal"], run_date=run_date
+            )
+
+            input_json = {"pairing": pairing, "mapping": mapping}
+
+            job_json = {"name": name, "app": app, "inputs": input_json, "tags": tags}
+            jobs.append(job_json)
+
+        # self.send_message(
+        #     """
+        #     Writing files to {file_path}.
+        #
+        #     Run Date: {run_date}
+        #     Beagle Version: {beagle_version}
+        #     """.format(
+        #         file_path=self.OUTPUT_DIR, run_date=run_date, beagle_version=beagle_version
+        #     )
+        # )
+
+        return [RunCreator(**job) for job in jobs]
```

Comment:
I think since this is mostly still based on the old TempoMPGen operator, we're going to be submitting every WES sample in the beagle database, which will amount to like 7000+ jobs each time we execute it.

I think the framework is here, but we'll need to update this so that we're running just new WES samples.

We may need to talk about adding some of the other text files we output here - mapping, pairing, unpaired, etc. files -  into their own tables in beagle somewhere.

Let's discuss this soon.